tests/assert_type/contrib/staticfiles/test_storage.py

```diff
@@ -0,0 +1,8 @@
+from django.contrib.staticfiles.storage import ConfiguredStorage, StaticFilesStorage, staticfiles_storage
+from typing_extensions import assert_type
+
+# The plugin can figure out what these are (but pyright can't):
+assert_type(staticfiles_storage, StaticFilesStorage)  # pyright: ignore[reportAssertTypeFailure]
+
+# what pyright thinks these are:
+assert_type(staticfiles_storage, ConfiguredStorage)  # mypy: ignore[assert-type]
```

Comment:
so, basically, right now `pyright` will infer `ConfiguredStorage` instead of `Any` and existing code can break, due to the fact that `ConfiguredStorage` is a `LazyObject` which does not really have right attributes. 

How can we fix that?

Reply:
It wasn't `Any` before, but it can be if that's better... I just looked and mypy can't have it's own type ignores, so I'm not sure how to ever make this work since mypy will always know more about the object than pyright does (even if it reports `Any` for pyright the assertion will fail, right?).